Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/DefaultLibrarySourceProvider.java

```diff
@@ -51,7 +51,10 @@ public boolean accept(File path, String name) {
                     Version version = new Version(fileName.substring(indexOfVersionSeparator + 1));
                     // If the file has a version, make sure it is compatible with the version we are looking for
                     if (indexOfVersionSeparator == libraryName.length() && requestedVersion == null || version.compatibleWith(requestedVersion)) {
-                        if (mostRecent == null || version.compareTo(mostRecent) > 0) {
+                        if (mostRecent == null ||
+                                ((version != null && version.isComparable()) &&
+                                        (mostRecent != null && mostRecent.isComparable()) &&
+                                        version.compareTo(mostRecent) > 0)) {
                             mostRecent = version;
```

Comment:
This is correct for the case that both are comparable, but if either or both are not comparable, then it should fall back to strict equality matching.

Reply:
Updated this. Thanks!